motor/mcpwm_foc.c

```diff
@@ -692,15 +692,28 @@ void mcpwm_foc_stop_pwm(bool is_second_motor) {
  *
  * @param dutyCycle
  * The duty cycle to use.
+ *
+ * @return
+ * The fault code
  */
-void mcpwm_foc_set_duty(float dutyCycle) {
+int mcpwm_foc_set_duty(float dutyCycle) {
```

Comment:
I think it is best if these don't return the fault code. The check is only done in the beginning anyway and if you are interested in that check you can just do it yourself. If the command itself can fail from the processing it does makes sense to return that result, but having it check some external state and returning that independently of what the command did just makes things messy and hard to follow.

Another reason for not adding extra function calls to these is that they might be called at a very high rate from threads that run control loops and we don't want to cause too much performance impact. If does not matter much now, but these small things add up over time in the code base.

Another thing: These commands bypass mc_interface, so you have to use them with a bit more responsibility in general. mc_interface usually does these high level checks for faults, timeout etc. and uses these functions accordingly.

Reply:
That is a fair point, I just checked and all these ones are only used from mc_interface, so I will remove the checks in here as they are duplicated.
```
void mcpwm_foc_set_duty(float dutyCycle);
void mcpwm_foc_set_duty_noramp(float dutyCycle);
void mcpwm_foc_set_pid_speed(float rpm);
void mcpwm_foc_set_pid_pos(float pos);
void mcpwm_foc_set_current(float current);
void mcpwm_foc_set_brake_current(float current);
void mcpwm_foc_set_handbrake(float current);
```

The openloop functions don't have a corresponding mc_interface command so we can either add one with the same checks as the other commands to keep things consistent in behavior. Or keep the checks in these functions. 
```
int mcpwm_foc_set_openloop(float current, float rpm);
int mcpwm_foc_set_openloop_phase(float current, float phase);
int mcpwm_foc_set_openloop_duty(float dutyCycle, float rpm);
int mcpwm_foc_set_openloop_duty_phase(float dutyCycle, float phase);
```

Reply:
I think it is best to add these to mc_interface and use mc_interface_try_input in the same way as with the other functions. Initially I treated these as specialized functions for my testing, but if they are generally useful it is best to run them via mc_interface with the same checks as the other functions.